src/main/java/org/commcare/suite/model/QueryPrompt.java

```diff
@@ -174,4 +176,21 @@ public boolean isSelect() {
         return getItemsetBinding() != null;
     }
 
+    // Evaluates required in the given eval context
+    public boolean isRequired(EvaluationContext ec) {
+        XPathExpression requiredCondition = required != null ? required.getTest() : null;
+        if (requiredCondition != null) {
+            return (Boolean)requiredCondition.eval(ec);
+        }
```

Comment:
nit: perhaps just preference but I prefer the way you coded it for the `getRequiredMessage` function
```suggestion
        if (required != null && required.getTest() != null) {
            return (Boolean) required.getTest().eval(ec);
        }
```